aten/src/ATen/native/Integration.cpp

```diff
@@ -69,4 +69,34 @@ Tensor trapz(const Tensor& y, double dx, int64_t dim) {
     return do_trapz(y, dx, dim);
 }
 
+Tensor do_cumulative_trapezoid(const Tensor& y, const Tensor& dx, int64_t dim) {
+  Tensor left = y.slice(dim, 0, -1);
+  Tensor right = y.slice(dim, 1);
+
+  return ((left + right) * dx).cumsum(dim) / 2.;
+}
+
+Tensor cumulative_trapezoid(const Tensor& y, const Tensor& x, int64_t dim) {
+  dim = maybe_wrap_dim(dim, y);
+  // asking for the integral with zero samples is a bit nonsensical,
```

Comment:
Also test that y and x are on the same device.

What happens if y and x are different dtypes?

Does this support multidimensional tensors, or should y and x be 1D?